Calculator.Logic/CalculatorLogic.cs

```diff
@@ -0,0 +1,323 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Calculator.Logic
+{
+    public class CalculatorLogic : ICalculatorLogic
+    {
+        private ICalculatorError _calculatorError;
+        private ICalculatorValidator _calculatorValidator;
+        private IStringToNumberConvertor _stringToNumberConverter;
+        private List<string> _terms;
+        private Operation _currentOperation;
+        private string _lastTerm;
+
+        public CalculatorLogic(ICalculatorError calculatorError, ICalculatorValidator calculatorValidator, IStringToNumberConvertor stringToNumberConverter)
+        {
+            _calculatorError = calculatorError;
+            _calculatorValidator = calculatorValidator;
+            _stringToNumberConverter = stringToNumberConverter;
+            _terms = new List<string>();
+        }
+       
+        public CalculatorResult DoMultipleTermOperation()
+        {
+            if (ShouldRepeatLastOperation(_currentOperation))
+            {
+                AddLastTermUsed(_lastTerm);
+            }
+            try
```

Comment:
this try catch should be one private method that is called from two different public methods (do single term operation, do multiple term operation)